app/src/main/java/org/mozilla/fenix/perf/RunBlockingCounter.kt

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// This class implements the alternative ways to invoke runBlocking with some
+// monitoring by wrapping the raw methods. This lint check tells us not to use the raw
+// methods so we suppress the check.
+@file:Suppress("MozillaRunBlockingCheck")
+
+package org.mozilla.fenix.perf
+
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.runBlocking
+import java.util.concurrent.atomic.AtomicInteger
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Counts the number of runBlocking calls made
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+object RunBlockingCounter {
+    var count = AtomicInteger(0)
+
+    fun incrementCounter() {
```

Comment:
nit: I think it's simpler to move this to be an extension function, e.g. `fun AtomicInteger.getAndIncrementNoOverflow()`. It lends itself to better testing (why isn't this method tested? :P), re-use, and more greatly discourages touching any code outside of the AtomicInteger, unlike a function in a companion object.